lib/screens/event/screen.dart

```diff
@@ -0,0 +1,159 @@
+                            flex: 9,
+                            child: SortDropdownMenu(
+                              preferences: 'eventsSortOption',
+                              streamController: dropdownMenuStreamController,
+                              value: optionVal.data,
+                              items: <String>[
+                                'Furthest',
+                                'Upcoming',
+                                'Most attendees',
+                                'Least attendees',
+                              ].map<DropdownMenuItem<String>>((String value) {
+                                Map<String, String> itemsMap = {
+                                  'Furthest': AppLocalizations.of(context)
+                                      .translate("furthest"),
+                                  'Upcoming': AppLocalizations.of(context)
+                                      .translate("upcoming"),
+                                  'Most attendees': AppLocalizations.of(context)
+                                      .translate("mostAttendees"),
+                                  'Least attendees':
+                                      AppLocalizations.of(context)
+                                          .translate("leastAttendees"),
+                                };
+                                return DropdownMenuItem<String>(
+                                  value: value,
+                                  child: Text(
+                                    itemsMap[value],
+                                  ),
+                                );
+                              }).toList(),
+                            ),
+                          ),
+                          Flexible(
+                            flex: 1,
+                            child: SearchButton(
+                              delegate: Search<Event>(context, "searchEvents"),
+                            ),
+                          ),
+                        ],
+                      ),
+                    ),
+                    Expanded(
+                      child: StreamBuilder<List<Event>>(
+                        stream: Function.apply(
+                            EventUtil.screenView, [optionVal.data]),
+                        builder: (BuildContext context,
+                            AsyncSnapshot<List<Event>> snapshot) {
+                          if (snapshot.hasError) return ErrorScreen();
+                          switch (snapshot.connectionState) {
+                            case ConnectionState.waiting:
+                              return Center(
+                                child: CircularProgressIndicator(),
+                              );
+                            default:
+                              if (snapshot.data == null) {
+                                // TODO: extraneous; change other files
+                                return ErrorScreen();
```

Comment:
why is this labeled as extraneous?

Reply:
because snapshot.hasError already checks if snapshot.data is null or not

Reply:
i don't think that's true https://api.flutter.dev/flutter/widgets/AsyncSnapshot/hasError.html

Reply:
snapshot.hasError asserts if (error != null). but if you click into 'error' in that page, it checks if data != null. if thats true, the error will be null, and then snapshot.hasData will be false.

Reply:
there's an ambiguity, though, for the case when data == null. Will error still be true then, or will it take on another value? and does snapshot.hasError catch that?

Reply:
it looks like even if data is null, error can be null?

Reply:
i tested this on no wifi connection and hasError catches snapshot.data when its null